<html>
<head>
  <title>VEP summary</title>
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
  <style>
    a.inactive {
      color: grey;
      pointer-events:none;
    }
  </style>
</head>
<body>
<div class='main_content container'><h1>G2P report</h1><p>Input and output files:</p><dl class='dl-horizontal'><dt>G2P list</dt><dd>G2P_testing/DDG2P.csv</dd><dt>Log directory</dt><dd>G2P_testing/con_log</dd><dt>HTML report</dt><dd>G2P_testing/repo.html</dd><dt>TXT report</dt><dd>G2P_testing/repo.txt</dd></dl><p>Counts:</p><dl class='dl-horizontal text-overflow'><dt>2262</dt><dd>G2P genes</dd><dt>7</dt><dd>G2P genes in input VCF file</dd><dt>4</dt><dd>G2P complete genes in input VCF file</dd></dl><h1>Summary of G2P complete genes per individual</h1><p>G2P complete gene: A sufficient number of variant hits for the observed allelic requirement in at least one of the gene's transcripts. Variants are filtered by frequency.</p><p>G2P gene was filtered using HGNC_id</p><p>Frequency thresholds and number of required variant hits for each allelic requirement:</p><table class='table table-bordered'><thead><tr><th>Allelic requirement</th><th>Frequency threshold for filtering</th><th>Variant counts by zygosity</th></tr></thead><tbody><tr><td>biallelic</td><td>0.005</td><td>HET >= 2 OR HOM >= 1</td></tr><tr><td>biallelic_PAR</td><td>0.005</td><td>HET >= 2 OR HOM >= 1</td></tr><tr><td>biallelic_autosomal</td><td>0.005</td><td>HET >= 2 OR HOM >= 1</td></tr><tr><td>hemizygous</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>mitochondrial</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic_PAR</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic_X_hem</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic_X_het</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic_Y_hem</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic_autosomal</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>x-linked dominant</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>x-linked over-dominance</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr></tbody></table><form>
<div class="checkbox">
  <label>
    <input class="target" type="checkbox"> Show only canonical transcript
  </label>
</div>
</form>
<ul>
<li><a class="not_canonical" href="#P1_ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive)_monoallelic_autosomal_ENST00000495106">P1 &gt; ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive) &gt; monoallelic_autosomal &gt; ENST00000495106</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive)_monoallelic_autosomal_ENST00000495852">P1 &gt; ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive) &gt; monoallelic_autosomal &gt; ENST00000495852</a> </li>
<li><a class="is_canonical" href="#P1_ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive)_monoallelic_autosomal_ENST00000370360">P1 &gt; ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive) &gt; monoallelic_autosomal &gt; ENST00000370360</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive)_monoallelic_autosomal_ENST00000463560">P1 &gt; ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive) &gt; monoallelic_autosomal &gt; ENST00000463560</a> </li>
</ul>
<ul>
<li><a class="is_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000244137">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000244137</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000590731">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000590731</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000397032">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000397032</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000590408">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000590408</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000436370">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000436370</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000588719">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000588719</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000609145">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000609145</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000588328">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000588328</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000590755">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000590755</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000651901">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000651901</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000651646">P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000651646</a> </li>
</ul>
<ul>
<li><a class="not_canonical" href="#P2_ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong)_biallelic_autosomal_ENST00000587734">P2 &gt; ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong) &gt; biallelic_autosomal &gt; ENST00000587734</a> </li>
<li><a class="is_canonical" href="#P2_ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong)_biallelic_autosomal_ENST00000294618">P2 &gt; ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong) &gt; biallelic_autosomal &gt; ENST00000294618</a> </li>
<li><a class="not_canonical" href="#P2_ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong)_biallelic_autosomal_ENST00000587656">P2 &gt; ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong) &gt; biallelic_autosomal &gt; ENST00000587656</a> </li>
</ul>
<ul>
<li><a class="is_canonical" href="#P3_ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive)_monoallelic_X_het_ENST00000374839">P3 &gt; ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive) &gt; monoallelic_X_het &gt; ENST00000374839</a> </li>
<li><a class="not_canonical" href="#P3_ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive)_monoallelic_X_het_ENST00000337990">P3 &gt; ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive) &gt; monoallelic_X_het &gt; ENST00000337990</a> </li>
</ul>
<ul>
<li><a class="is_canonical" href="#P3_ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive)_monoallelic_X_hem_ENST00000374839">P3 &gt; ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive) &gt; monoallelic_X_hem &gt; ENST00000374839</a> </li>
<li><a class="not_canonical" href="#P3_ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive)_monoallelic_X_hem_ENST00000337990">P3 &gt; ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive) &gt; monoallelic_X_hem &gt; ENST00000337990</a> </li>
</ul>
<div class="not_canonical"><h3><a name="P1_ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive)_monoallelic_autosomal_ENST00000495106"></a>P1 &gt; ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive) &gt; monoallelic_autosomal &gt; ENST00000495106 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>1:92267915-92267915 G/A</td><td>1_92267915_G/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs749444788">rs749444788</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>stop_gained,NMD_transcript_variant,splice_region_variant</td><td>NA</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>3.991e-06</td><td>6.328e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000495106.5:c.1096C>T</td><td>ENSP00000436829.1:p.Gln366Ter</td><td>-</td></tr>
<tr><td>1:92266456-92266456 T/A</td><td>1_92266456_T/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1397019423">rs1397019423</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>missense_variant,NMD_transcript_variant</td><td>NA</td><td class="danger">deleterious(0.0)</td><td class="success">benign(0.444)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>4.012e-06</td><td></td><td></td><td></td><td></td><td></td><td>8.86e-06</td><td></td><td></td><td>ENST00000495106.5:c.1177A>T</td><td>ENSP00000436829.1:p.Asn393Tyr</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive)_monoallelic_autosomal_ENST00000495852"></a>P1 &gt; ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive) &gt; monoallelic_autosomal &gt; ENST00000495852 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>1:92267915-92267915 G/A</td><td>1_92267915_G/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs749444788">rs749444788</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>stop_gained,splice_region_variant</td><td>NA</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>3.991e-06</td><td>6.328e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000495852.6:c.361C>T</td><td>ENSP00000469157.2:p.Gln121Ter</td><td>-</td></tr>
<tr><td>1:92266456-92266456 T/A</td><td>1_92266456_T/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1397019423">rs1397019423</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>missense_variant</td><td>NA</td><td class="danger">deleterious(0.0)</td><td class="warning">possibly damaging(0.458)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>4.012e-06</td><td></td><td></td><td></td><td></td><td></td><td>8.86e-06</td><td></td><td></td><td>ENST00000495852.6:c.400A>T</td><td>ENSP00000469157.2:p.Asn134Tyr</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="is_canonical"><h3><a name="P1_ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive)_monoallelic_autosomal_ENST00000370360"></a>P1 &gt; ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive) &gt; monoallelic_autosomal &gt; ENST00000370360 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>1:92267915-92267915 G/A</td><td>1_92267915_G/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs749444788">rs749444788</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>stop_gained,splice_region_variant</td><td>NA</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>3.991e-06</td><td>6.328e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000370360.8:c.1096C>T</td><td>ENSP00000359385.3:p.Gln366Ter</td><td>NM_001319683.2,NM_053274.3</td></tr>
<tr><td>1:92266456-92266456 T/A</td><td>1_92266456_T/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1397019423">rs1397019423</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>missense_variant</td><td>NA</td><td class="danger">deleterious(0.0)</td><td class="success">benign(0.12)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>4.012e-06</td><td></td><td></td><td></td><td></td><td></td><td>8.86e-06</td><td></td><td></td><td>ENST00000370360.8:c.1177A>T</td><td>ENSP00000359385.3:p.Asn393Tyr</td><td>NM_001319683.2,NM_053274.3</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive)_monoallelic_autosomal_ENST00000463560"></a>P1 &gt; ENSG00000174842(GLMN,VMGLOM HGNC:14373;  G2P confidence category:definitive) &gt; monoallelic_autosomal &gt; ENST00000463560 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>1:92267915-92267915 G/A</td><td>1_92267915_G/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs749444788">rs749444788</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>stop_gained,splice_region_variant</td><td>NA</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>3.991e-06</td><td>6.328e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000463560.1:c.466C>T</td><td>ENSP00000468973.1:p.Gln156Ter</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="is_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000244137"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000244137 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000244137.12:c.504-1G>T</td><td>NA</td><td>NM_000285.4</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000590731"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000590731 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant,non_coding_transcript_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000590731.6:n.179-1G>T</td><td>NA</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000397032"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000397032 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000397032.8:c.504-1G>T</td><td>NA</td><td>NM_001166056.2</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000590408"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000590408 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant,NMD_transcript_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000590408.1:c.*95-1G>T</td><td>NA</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000436370"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000436370 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000436370.7:c.312-1G>T</td><td>NA</td><td>NM_001166057.2</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000588719"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000588719 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant,non_coding_transcript_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000588719.5:n.139-1G>T</td><td>NA</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000609145"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000609145 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant,non_coding_transcript_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000609145.5:n.194-1G>T</td><td>NA</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000588328"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000588328 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000588328.6:c.493-1G>T</td><td>NA</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000590755"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000590755 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant,NMD_transcript_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000590755.6:c.111-1G>T</td><td>NA</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000651901"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000651901 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000651901.1:c.500-1G>T</td><td>NA</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive)_biallelic_autosomal_ENST00000651646"></a>P1 &gt; ENSG00000124299(PEPD HGNC:8840;  G2P confidence category:definitive) &gt; biallelic_autosomal &gt; ENST00000651646 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:33478091-33478091 C/A</td><td>19_33478091_C/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs542228812">rs542228812</a></td><td>HOM</td><td>biallelic_autosomal</td><td>splice_acceptor_variant</td><td>likely_pathogenic</td><td>NA</td><td>NA</td><td>no</td><td>no</td><td>0.0008</td><td></td><td></td><td></td><td></td><td>4.105e-06</td><td>6.784e-05</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000651646.1:c.502-1G>T</td><td>NA</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P2_ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong)_biallelic_autosomal_ENST00000587734"></a>P2 &gt; ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong) &gt; biallelic_autosomal &gt; ENST00000587734 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:11199515-11199515 G/C</td><td>19_11199515_G/C</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1332239671">rs1332239671</a></td><td>HOM</td><td>biallelic_autosomal</td><td>missense_variant</td><td>NA</td><td>NA</td><td class="success">benign(0.0)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>4.943e-06</td><td></td><td></td><td></td><td></td><td></td><td>1.133e-05</td><td></td><td></td><td>ENST00000587734.1:c.100C>G</td><td>ENSP00000468291.1:p.Pro34Ala</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="is_canonical"><h3><a name="P2_ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong)_biallelic_autosomal_ENST00000294618"></a>P2 &gt; ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong) &gt; biallelic_autosomal &gt; ENST00000294618 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:11211872-11211872 T/C</td><td>19_11211872_T/C</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs779305788">rs779305788</a></td><td>HOM</td><td>biallelic_autosomal</td><td>missense_variant</td><td>NA</td><td class="success">tolerated(0.56)</td><td class="success">benign(0.007)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>2.495e-05</td><td></td><td></td><td></td><td>0.0003544</td><td></td><td></td><td></td><td></td><td>ENST00000294618.12:c.4655A>G</td><td>ENSP00000294618.6:p.Gln1552Arg</td><td>NM_020812.4,NM_001367830.1</td></tr>
<tr><td>19:11243373-11243373 G/A</td><td>19_11243373_G/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs989512077">rs989512077</a></td><td>HOM</td><td>biallelic_autosomal</td><td>missense_variant</td><td>NA</td><td class="success">tolerated(0.09)</td><td class="success">benign(0.04)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td></td><td>ENST00000294618.12:c.1271C>T</td><td>ENSP00000294618.6:p.Ala424Val</td><td>NM_020812.4,NM_001367830.1</td></tr>
<tr><td>19:11199515-11199515 G/C</td><td>19_11199515_G/C</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1332239671">rs1332239671</a></td><td>HOM</td><td>biallelic_autosomal</td><td>missense_variant</td><td>NA</td><td>tolerated - low confidence(0.48)</td><td class="success">benign(0.005)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>4.943e-06</td><td></td><td></td><td></td><td></td><td></td><td>1.133e-05</td><td></td><td></td><td>ENST00000294618.12:c.6126C>G</td><td>ENSP00000294618.6:p.Phe2042Leu</td><td>NM_020812.4,NM_001367830.1</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P2_ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong)_biallelic_autosomal_ENST00000587656"></a>P2 &gt; ENSG00000130158(DOCK6 HGNC:19189;  G2P confidence category:strong) &gt; biallelic_autosomal &gt; ENST00000587656 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>19:11211872-11211872 T/C</td><td>19_11211872_T/C</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs779305788">rs779305788</a></td><td>HOM</td><td>biallelic_autosomal</td><td>missense_variant</td><td>NA</td><td class="success">tolerated(0.51)</td><td class="warning">possibly damaging(0.5)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>2.495e-05</td><td></td><td></td><td></td><td>0.0003544</td><td></td><td></td><td></td><td></td><td>ENST00000587656.5:c.2522A>G</td><td>ENSP00000468638.1:p.Gln841Arg</td><td>-</td></tr>
<tr><td>19:11199515-11199515 G/C</td><td>19_11199515_G/C</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1332239671">rs1332239671</a></td><td>HOM</td><td>biallelic_autosomal</td><td>missense_variant</td><td>NA</td><td>tolerated - low confidence(0.45)</td><td class="success">benign(0.009)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>4.943e-06</td><td></td><td></td><td></td><td></td><td></td><td>1.133e-05</td><td></td><td></td><td>ENST00000587656.5:c.3993C>G</td><td>ENSP00000468638.1:p.Phe1331Leu</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="is_canonical"><h3><a name="P3_ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive)_monoallelic_X_het_ENST00000374839"></a>P3 &gt; ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive) &gt; monoallelic_X_het &gt; ENST00000374839 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>X:64919134-64919134 C/T</td><td>X_64919134_C/T</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs756615930">rs756615930</a></td><td>HET</td><td>monoallelic_X_het,monoallelic_X_hem</td><td>missense_variant</td><td>likely_benign</td><td class="success">tolerated(0.21)</td><td class="danger">probably damaging(0.994)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>2.275e-05</td><td></td><td>3.783e-05</td><td></td><td>7.491e-05</td><td></td><td>2.545e-05</td><td></td><td></td><td>ENST00000374839.8:c.469G>A</td><td>ENSP00000363972.3:p.Ala157Thr</td><td>NM_018684.4</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P3_ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive)_monoallelic_X_het_ENST00000337990"></a>P3 &gt; ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive) &gt; monoallelic_X_het &gt; ENST00000337990 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>X:64919134-64919134 C/T</td><td>X_64919134_C/T</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs756615930">rs756615930</a></td><td>HET</td><td>monoallelic_X_het,monoallelic_X_hem</td><td>missense_variant</td><td>likely_benign</td><td class="success">tolerated(0.28)</td><td class="danger">probably damaging(0.994)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>2.275e-05</td><td></td><td>3.783e-05</td><td></td><td>7.491e-05</td><td></td><td>2.545e-05</td><td></td><td></td><td>ENST00000337990.2:c.400G>A</td><td>ENSP00000338650.2:p.Ala134Thr</td><td>NM_001243804.2,NM_001178032.3</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="is_canonical"><h3><a name="P3_ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive)_monoallelic_X_hem_ENST00000374839"></a>P3 &gt; ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive) &gt; monoallelic_X_hem &gt; ENST00000374839 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>X:64919134-64919134 C/T</td><td>X_64919134_C/T</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs756615930">rs756615930</a></td><td>HET</td><td>monoallelic_X_het,monoallelic_X_hem</td><td>missense_variant</td><td>likely_benign</td><td class="success">tolerated(0.21)</td><td class="danger">probably damaging(0.994)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>2.275e-05</td><td></td><td>3.783e-05</td><td></td><td>7.491e-05</td><td></td><td>2.545e-05</td><td></td><td></td><td>ENST00000374839.8:c.469G>A</td><td>ENSP00000363972.3:p.Ala157Thr</td><td>NM_018684.4</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P3_ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive)_monoallelic_X_hem_ENST00000337990"></a>P3 &gt; ENSG00000126970(KIAA1166,MCS,MRXS4,WWS,ZC4H2 HGNC:24931;  G2P confidence category:definitive) &gt; monoallelic_X_hem &gt; ENST00000337990 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AFR">AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:AMR">AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EAS">EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:EUR">EUR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="1000GENOMES:phase_3:SAS">SAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>X:64919134-64919134 C/T</td><td>X_64919134_C/T</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs756615930">rs756615930</a></td><td>HET</td><td>monoallelic_X_het,monoallelic_X_hem</td><td>missense_variant</td><td>likely_benign</td><td class="success">tolerated(0.28)</td><td class="danger">probably damaging(0.994)</td><td>no</td><td>no</td><td></td><td></td><td></td><td></td><td></td><td>2.275e-05</td><td></td><td>3.783e-05</td><td></td><td>7.491e-05</td><td></td><td>2.545e-05</td><td></td><td></td><td>ENST00000337990.2:c.400G>A</td><td>ENSP00000338650.2:p.Ala134Thr</td><td>NM_001243804.2,NM_001178032.3</td></tr>
</tbody>
</TABLE>
</div>
</div>

</div>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  <script type="text/javascript" src="https://www.google.com/jsapi"></script>
  <script>
    $( "input[type=checkbox]" ).on( "click", function(){
      if ($('.target').is(':checked')) {
        $( "div.not_canonical" ).hide();
        $("a.not_canonical").addClass("inactive");
      } else {
        $( "div.not_canonical" ).show();
        $("a.not_canonical").removeClass("inactive");
      }
    } );
  $(document).ready(function(){
    $('[data-toggle="tooltip"]').tooltip(); 
  });
  </script>

</body>
</html>
