<html>
<head>
  <title>VEP summary</title>
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
  <style>
    a.inactive {
      color: grey;
      pointer-events:none;
    }
  </style>
</head>
<body>
<div class='main_content container'><h1>G2P report</h1><p>Input and output files:</p><dl class='dl-horizontal'><dt>G2P list</dt><dd>/nfs/production/flicek/ensembl/variation/G2P/panels/DDG2P.csv</dd><dt>Log directory</dt><dd>/hps/nobackup/flicek/ensembl/variation/dlemos/G2P/test_plugin/example/g2p_log_dir_2021_12_14_13_14_13</dd><dt>HTML report</dt><dd>report.html</dd><dt>TXT report</dt><dd>report.txt</dd></dl><p>Counts:</p><dl class='dl-horizontal text-overflow'><dt>2233</dt><dd>G2P genes</dd><dt>1</dt><dd>G2P genes in input VCF file</dd><dt>1</dt><dd>G2P complete genes in input VCF file</dd></dl><h1>Summary of G2P complete genes per individual</h1><p>G2P complete gene: A sufficient number of variant hits for the observed allelic requirement in at least one of the gene's transcripts. Variants are filtered by frequency.</p><p>Frequency thresholds and number of required variant hits for each allelic requirement:</p><table class='table table-bordered'><thead><tr><th>Allelic requirement</th><th>Frequency threshold for filtering</th><th>Variant counts by zygosity</th></tr></thead><tbody><tr><td>biallelic</td><td>0.005</td><td>HET >= 2 OR HOM >= 1</td></tr><tr><td>biallelic_autosomal</td><td>0.005</td><td>HET >= 2 OR HOM >= 1</td></tr><tr><td>hemizygous</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic_X_hem</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic_X_het</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>monoallelic_autosomal</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>x-linked dominant</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr><tr><td>x-linked over-dominance</td><td>0.0001</td><td>HET >= 1 OR HOM >= 1</td></tr></tbody></table><form>
<div class="checkbox">
  <label>
    <input class="target" type="checkbox"> Show only canonical transcript
  </label>
</div>
</form>
<ul>
<li><a class="not_canonical" href="#P1_ENSG00000174842(GLMN,VMGLOM)_monoallelic_autosomal_ENST00000495106">P1 &gt; ENSG00000174842(GLMN,VMGLOM) &gt; monoallelic_autosomal &gt; ENST00000495106</a> </li>
<li><a class="not_canonical" href="#P1_ENSG00000174842(GLMN,VMGLOM)_monoallelic_autosomal_ENST00000495852">P1 &gt; ENSG00000174842(GLMN,VMGLOM) &gt; monoallelic_autosomal &gt; ENST00000495852</a> </li>
<li><a class="is_canonical" href="#P1_ENSG00000174842(GLMN,VMGLOM)_monoallelic_autosomal_ENST00000370360">P1 &gt; ENSG00000174842(GLMN,VMGLOM) &gt; monoallelic_autosomal &gt; ENST00000370360</a> </li>
</ul>
<div class="not_canonical"><h3><a name="P1_ENSG00000174842(GLMN,VMGLOM)_monoallelic_autosomal_ENST00000495106"></a>P1 &gt; ENSG00000174842(GLMN,VMGLOM) &gt; monoallelic_autosomal &gt; ENST00000495106 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>1:92266456-92266456 T/A</td><td>1_92266456_T/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1397019423">rs1397019423</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>missense_variant,NMD_transcript_variant</td><td>NA</td><td class="danger">deleterious(0.0)</td><td class="success">benign(0.444)</td><td>no</td><td>no</td><td>4.012e-06</td><td></td><td></td><td></td><td></td><td></td><td>8.86e-06</td><td></td><td></td><td>ENST00000495106.5:c.1177A>T</td><td>ENSP00000436829.1:p.Asn393Tyr</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="not_canonical"><h3><a name="P1_ENSG00000174842(GLMN,VMGLOM)_monoallelic_autosomal_ENST00000495852"></a>P1 &gt; ENSG00000174842(GLMN,VMGLOM) &gt; monoallelic_autosomal &gt; ENST00000495852 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>1:92266456-92266456 T/A</td><td>1_92266456_T/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1397019423">rs1397019423</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>missense_variant</td><td>NA</td><td class="danger">deleterious(0.0)</td><td class="warning">possibly damaging(0.458)</td><td>no</td><td>no</td><td>4.012e-06</td><td></td><td></td><td></td><td></td><td></td><td>8.86e-06</td><td></td><td></td><td>ENST00000495852.6:c.400A>T</td><td>ENSP00000469157.2:p.Asn134Tyr</td><td>-</td></tr>
</tbody>
</TABLE>
</div>
</div>
<div class="is_canonical"><h3><a name="P1_ENSG00000174842(GLMN,VMGLOM)_monoallelic_autosomal_ENST00000370360"></a>P1 &gt; ENSG00000174842(GLMN,VMGLOM) &gt; monoallelic_autosomal &gt; ENST00000370360 <a title="Back to Top" data-toggle="tooltip" href='#top'><span class="glyphicon glyphicon-arrow-up" aria-hidden="true"></span></a></h3>
<div class="table-responsive" style="width:100%">
<TABLE  class="table table-bordered table-condensed" style="margin-left: 2em"><thead>
<tr><th>Variant location and alleles (REF/ALT)</th><th>Variant name (* indicates that variant is on variant include list)</th><th>Existing name</th><th>Zygosity</th><th>All allelic requirements from G2P DB</th><th>Consequence types</th><th>ClinVar annotation</th><th>SIFT</th><th>PolyPhen</th><th>Novel variant</th><th>Has been failed by Ensembl</th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database:Total">gnomAD</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:African/African American">gnomAD_AFR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Latino/Admixed American">gnomAD_AMR</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Ashkenazi Jewish">gnomAD_ASJ</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:East Asian">gnomAD_EAS</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Finnish">gnomAD_FIN</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Non-Finnish European">gnomAD_NFE</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:Other (population not assigned)">gnomAD_OTH</a></th><th><a style="cursor: pointer" data-placement="top" data-toggle="tooltip" data-container="body" title="Genome Aggregation Database exomes v2.1:South Asian">gnomAD_SAS</a></th><th>HGVS transcript</th><th>HGVS protein</th><th>RefSeq IDs</th></tr>
</thead>
<tbody>
<tr><td>1:92266456-92266456 T/A</td><td>1_92266456_T/A</td><td><a href="https://ensembl.org/Homo_sapiens/Variation/Explore?v=rs1397019423">rs1397019423</a></td><td>HOM</td><td>monoallelic_autosomal</td><td>missense_variant</td><td>NA</td><td class="danger">deleterious(0.0)</td><td class="success">benign(0.12)</td><td>no</td><td>no</td><td>4.012e-06</td><td></td><td></td><td></td><td></td><td></td><td>8.86e-06</td><td></td><td></td><td>ENST00000370360.8:c.1177A>T</td><td>ENSP00000359385.3:p.Asn393Tyr</td><td>NM_001319683.2,NM_053274.3</td></tr>
</tbody>
</TABLE>
</div>
</div>

</div>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  <script type="text/javascript" src="https://www.google.com/jsapi"></script>
  <script>
    $( "input[type=checkbox]" ).on( "click", function(){
      if ($('.target').is(':checked')) {
        $( "div.not_canonical" ).hide();
        $("a.not_canonical").addClass("inactive");
      } else {
        $( "div.not_canonical" ).show();
        $("a.not_canonical").removeClass("inactive");
      }
    } );
  $(document).ready(function(){
    $('[data-toggle="tooltip"]').tooltip(); 
  });
  </script>

</body>
</html>
